js/conversation_controller.js

```diff
@@ -29,11 +29,10 @@
         },
         updateUnreadCount: function(model, count) {
             var prev = model.previous('unreadCount') || 0;
-            if (count < prev) { // decreased
-                var newUnreadCount = storage.get("unreadCount", 0) - (prev - count);
-                setUnreadCount(newUnreadCount);
-                storage.put("unreadCount", newUnreadCount);
-            }
+            var newUnreadCount = storage.get("unreadCount", 0) - (prev - count);
+            setUnreadCount(newUnreadCount);
+            storage.remove("unreadCount");
```

Comment:
I tried calling the methods from the console of the background page and noticed that simply overwriting didn't change the value.

By calling "get" for the second time, we can see that the value remains unchanged:
![screenshot from 2015-11-14 01 09 24](https://cloud.githubusercontent.com/assets/9749079/11159931/cffc63c4-8a6c-11e5-8b62-9d18336cdad8.png)